src/api/functional_tests.js

```diff
@@ -1,306 +1,272 @@
+      if (workflow.conclusion !== "cancelled") {
+        const workflow_run = {
+          id: workflow.id,
+          workflow_id: workflow.workflow_id,
+          check_suite_id: workflow.check_suite_id,
+          name: workflow.name,
+          started: workflow.run_started_at,
+          title: workflow.display_title,
+          status: workflow.status,
+          conclusion: workflow.conclusion,
+          branch: workflow.head_branch,
+          actor_email: workflow.head_commit.author.email,
+          actor_name: workflow.head_commit.author.name,
+          actor_message: workflow.head_commit.message,
+          actor_commit_time: workflow.head_commit.timestamp,
+          sha: workflow.sha,
+          event: workflow.event,
+          node_id: workflow.node_id,
+          url: workflow.url,
+          html_url: workflow.html_url,
+          created: workflow.created_at,
+          updated: workflow.updated_at,
+          run_attempt: workflow.run_attempt,
+          referenced_wf: workflow.referenced_workflows[0],
+          triggering_actor: workflow.triggering_actor.login,
+          jobs_url: workflow.jobs_url,
+          logs_url: workflow.logs_url,
+          workflow_url: workflow.workflow_url,
+        };
+
+        return_data.push(workflow_run);
+      }
     }
+  } catch (err) {
+    console.log("Unable to fetch getSingleWorkflow", err);
+    return;
+  }
 
-    // save to json file
-    return return_data;
+  return return_data;
 }
 
 async function _getWorkflowCheckRuns(checkSuiteId) {
-    const workflowCheckRunsApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/check-suites/${checkSuiteId}/check-runs`    
-    let return_data = [];
-    
-    try {
-        const { data } = await axios.get(workflowCheckRunsApi)
-        for(let i=0; i < data.check_runs.length; i++){
-            const check_run = data.check_runs[i]
-            if(check_run.name !== 'Build Test Client'){
-                const run = {
-                    id: check_run.id,
-                    name: check_run.name,
-                    status: check_run.status,
-                    conclusion: check_run.conclusion,
-                    head_sha: check_run.head_sha,
-                    started: check_run.started_at,
-                    completed: check_run.completed_at,
-                    annotations: check_run.output.annotations_count
-                }
-    
-                return_data.push(run)
-            }
-        }        
-    } catch(err){
-        console.log('Unable to fetch getSingleWorkflow', err);
-        return
+  const workflowCheckRunsApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/check-suites/${checkSuiteId}/check-runs`;
+  let return_data = [];
+
+  try {
+    const { data } = await axios.get(workflowCheckRunsApi);
+    for (let i = 0; i < data.check_runs.length; i++) {
+      const check_run = data.check_runs[i];
+      if (check_run.name !== "Build Test Client") {
```

Comment:
why?

Reply:
build test client is not a functional test